skeleton/src/main/java/com/dropbox/kaiken/skeleton/initializers/AppServicesInitializer.kt

```diff
@@ -0,0 +1,44 @@
+package com.dropbox.kaiken.skeleton.initializers
+
+import com.dropbox.common.inject.AppScope
+import com.dropbox.kaiken.skeleton.scoping.SingleIn
+import com.squareup.anvil.annotations.ContributesMultibinding
+import com.squareup.anvil.annotations.ContributesTo
+import javax.inject.Inject
+
+/**
+ * Interface to use when you have a component which you want to be called
+ * immediately at app startup.  Be mindful anything done here will block the app from launching.
+ *
+ * @ContributesMultibinding(SdkScope::class)
+ * @SingleIn(SdkScope::class)
+ * class FooInitializer @Inject constructor(
+ *     private val foo: Foo,
+ *     private val bar: Bar,
+ * ): AppInitializer {
+ *    override fun init() {
+ *       foo.setup(bar)
+ *    }
+ * }
+ *
+ */
+interface AppServicesInitializer {
+    /**
+     * Called when the app is started
+     */
+    fun init()
```

Comment:
Currently, we have a teardown, but no init.  We also don't call the teardown anywhere, and I am not sure where we would? 

Thoughts on keeping it or just remove it?

Reply:
Yes, we should have a tear down function, and consider having another interface / providers for teardown.

Reply:
I think we should definitely have an init function. I'm not sure what are we thinking of terms of delay initialization (e.g. UploadQueue or Offline Download Queue). But it'd probably would go somewhere here.